R/error-checks.R

```diff
@@ -486,3 +500,71 @@ validateHasUniqueValues <- function(data, dataName = "dataset", na.rm = TRUE, nu
   }
   stop(messages$errorHasNoUniqueValues(data, dataName, na.rm))
 }
+
+validateIsIncludedInDataset <- function(columnNames, dataset, datasetName = NULL, nullAllowed = FALSE, logFolder = NULL) {
+  if (nullAllowed && is.null(columnNames)) {
+    return()
+  }
+  if (isIncluded(columnNames, names(dataset))) {
+    return()
+  }
+  if (is.null(logFolder)) {

```

Comment:
hum....should not the logErrorThenStop be responsifile for this (e.g. checkign the log folder) This is noise for the caller and something we will forget

Reply:
I totally agree. This is one of the things I was thinking when I said the error handling should be improved.
Currently, `logErrorThenStop` has a default log folder of `getwd()` meaning that if used without a logFolder it can possible create a new log file somewhere out of the workflow folder.
And since some of the error checkings happen prior the workflow creation, we need to discuss if we want a log for them or directly print the error on the console.

Reply:
ok . Can you create an issue for this (if it's not there already). I want the logging mechanism to be easy and transparent to use